pkg/cri/sbserver/sandbox_status.go

```diff
@@ -42,12 +43,32 @@ func (c *criService) PodSandboxStatus(ctx context.Context, r *runtime.PodSandbox
 		return nil, fmt.Errorf("failed to get sandbox controller: %w", err)
 	}
 
+	var (
+		createdAt time.Time
+		state     string
+		info      map[string]string
+	)
 	cstatus, err := controller.Status(ctx, sandbox.ID, r.GetVerbose())
 	if err != nil {
-		return nil, fmt.Errorf("failed to query controller status: %w", err)
+		// If the shim died unexpectedly (segfault etc.) let's set the state as
+		// NOTREADY and not just error out to make k8s and clients like crictl
+		// happy. If we get back ErrNotFound from controller.Status above while
+		// we're using the shim-mode controller, this is a decent indicator it
+		// exited unexpectedly. We can use the fact that we successfully retrieved
+		// the sandbox object from the store above to tell that this is true, otherwise
+		// if we followed the normal k8s convention of StopPodSandbox -> RemovePodSandbox,
+		// we wouldn't have that object in the store anymore.
+		if !errdefs.IsNotFound(err) {
+			return nil, fmt.Errorf("failed to query controller status: %w", err)
+		}
+		state = "SANDBOX_NOTREADY"
```

Comment:
```suggestion
		state = runtime.PodSandboxState_SANDBOX_READY
```

Reply:
The const is an integer, toCRISandboxStatus takes in a string, we can do `runtime.PodSandboxState_SANDBOX_NOTREADY.String()` though. Looks like `toCRISandboxStatus` sets the state to NOTREADY internally if we can't find a mapping for the passed in state now that I'm looking deeper, but I'd prefer to pass in NOTREADY explicitly as it's much clearer at the callsite here what we want to happen